jsapp/js/components/languages/languageSelector.stories.tsx

```diff
@@ -1,15 +1,102 @@
+  languagesResponsePage2nd,
+  languagesResponseQuerySwed,
+} from './languagesListStore.mock'
 
-// TODO: somehow mock the `languagesStore` here, so that there are means to play with this properly :)
 const meta: Meta<typeof LanguageSelector> = {
   title: 'Components/LanguageSelector',
   component: LanguageSelector,
   argTypes: {},
+  parameters: {
+    msw: {
+      handlers: [
+        http.get('/api/v2/languages/', (info) => {
+          if (info.request.url.endsWith('limit=100&offset=100')) {
+            return HttpResponse.json(languagesResponsePage2nd)
+          } else if (info.request.url.endsWith('q=swed')) {
+            return HttpResponse.json(languagesResponseQuerySwed)
+          } else {
+            return HttpResponse.json(languagesResponsePage1st)
+          }
+        }),
+        http.get('/environment/', () => HttpResponse.json(environmentResponse)),
+      ],
+    },
+  },
 }
 
 export default meta
+type StoryArgs = LanguageSelectorProps
+type Story = StoryObj<typeof LanguageSelector> & { args?: StoryArgs }
 
-type Story = StoryObj<typeof LanguageSelector>
+export const Default: Story = {}
 
-export const Primary: Story = {}
+/**
+ * This test is searching for "Swedish" language, selecting it, and then undoing both the selection and the search.
+ */
+export const TestSearchAndSelection: Story = {
+  args: {
+    onLanguageChange: fn(),
+  },
+  play: async ({ args, canvasElement, step }) => {
+    const canvas = within(canvasElement)
+
+    await step(
+      'Wait for languages list to be ready and verify that the "Arabic" language is present in the initial list.',
+      async () => {
+        await sleep(2000)
+        const noSearchItem = await canvas.findByText(/^Arabic/)
+        await expect(noSearchItem).toBeInTheDocument()
+      },
+    )
+
+    let searchResultItem: HTMLElement
+    await step(
+      'Type "swed" to find "Swedish" language. Wait for API response and verify that the language is present in the list.',
+      async () => {
+        const searchInput = await canvas.findByRole('searchbox')
+        await userEvent.type(searchInput, 'swed')
+        await sleep(2000)
+        searchResultItem = await canvas.findByText(/^Swedish/)
+        await expect(searchResultItem).toBeInTheDocument()
+      },
+    )
+
+    let selectedLanguage: HTMLElement
+    await step(
+      'Click the result to select the language. Search result should be cleared, and the selected language should be displayed.',
+      async () => {
+        await userEvent.click(searchResultItem)
+        await expect(searchResultItem).not.toBeInTheDocument()
+        selectedLanguage = await canvas.findByText(/^Swedish/)
+        await expect(selectedLanguage).toBeInTheDocument()
+        // Verify that the onLanguageChange callback have been called (i.e. parent component is informed which language
+        // was selected)
+        await expect((args as StoryArgs).onLanguageChange).toHaveBeenCalledTimes(1)
+      },
+    )
```

Comment:
I just tried it out, although undocumented, but `step` can be nested! :tada: That means, we can use a proper testing verbiage, in the form of
```
- action: "do this"
  - assertion: "A should be X"
  - assertion: "B should be Y"
  - ...
- ...  
```
e.g. in this case

```suggestion
    let selectedLanguage: HTMLElement
    await step('select the Swedish language by clicking on it', async () => {
      await userEvent.click(searchResultItem)

      await step('the `onLanguageChange` should be called with {code: "sv"}', async () => {
        await expect(args.onLanguageChange).toHaveBeenCalledWith(expect.objectContaining({ code: 'sv' }))
      })
      await step('the search result should be cleared', async () => {
        await expect(searchResultItem).not.toBeInTheDocument()
      })
      await step('the selected language should be displayed', async () => {
        selectedLanguage = await canvas.findByText(/^Swedish/)
        await expect(selectedLanguage).toBeInTheDocument()
      })
    })
```